x-pack/plugins/apm/server/routes/mobile/get_mobile_main_statistics_by_field.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  termQuery,
+  kqlQuery,
+  rangeQuery,
+} from '@kbn/observability-plugin/server';
+import { ProcessorEvent } from '@kbn/observability-plugin/common';
+import {
+  SERVICE_NAME,
+  SESSION_ID,
+  TRANSACTION_DURATION,
+  ERROR_TYPE,
+  AGENT_NAME,
+} from '../../../common/es_fields/apm';
+import { environmentQuery } from '../../../common/utils/environment_query';
+import { APMEventClient } from '../../lib/helpers/create_es_client/create_apm_event_client';
+import { getLatencyValue } from '../../lib/helpers/latency_aggregation_type';
+import { LatencyAggregationType } from '../../../common/latency_aggregation_types';
+import { calculateThroughputWithRange } from '../../lib/helpers/calculate_throughput';
+
+interface Props {
+  kuery: string;
+  apmEventClient: APMEventClient;
+  serviceName: string;
+  environment: string;
+  start: number;
+  end: number;
+  field: string;
+}
+
+export interface MobileMainStatisticsResponse {
+  mainStatistics: Array<{
+    name: string | number;
+    latency: number | null;
+    throughput: number;
+    crashRate: number;
+  }>;
+}
+
+export async function getMobileMainStatisticsByField({
+  kuery,
+  apmEventClient,
+  serviceName,
+  environment,
+  start,
+  end,
+  field,
+}: Props) {
+  const response = await apmEventClient.search(
+    `get_mobile_transaction_events_main_statistics_by_field`,
+    {
+      apm: {
+        events: [ProcessorEvent.transaction, ProcessorEvent.error],
```

Comment:
Can you use `sources` here as well? You'll need to add an `ApmDocumentType` for errors.

Reply:
It looks like this should actually be two requests, one for transaction events for latency and sessions, and then a separate one for errors to get the number of crashes?

Reply:
Addressed here https://github.com/elastic/kibana/pull/153044/commits/a79ebaa66ebd32c6ef351b2b7eaddfc3acd60c6c